packages/medusa/src/strategies/product/import.ts

```diff
@@ -0,0 +1,772 @@
+  protected readonly productVariantRepo_: typeof ProductVariantRepository
+
+  protected readonly csvParser_: CsvParser<
+    ProductImportCsvSchema,
+    Record<string, string>,
+    Record<string, string>
+  >
+
+  constructor(container) {
+    super(container)
+
+    const {
+      batchJobService,
+      productService,
+      productRepository,
+      productOptionRepository,
+      productVariantService,
+      productVariantRepository,
+      shippingProfileService,
+      regionRepository,
+      fileService,
+      redisClient,
+      manager,
+    } = container
+
+    this.csvParser_ = new CsvParser(container, CSVSchema)
+
+    this.manager_ = manager
+    this.redisClient_ = redisClient
+    this.fileService_ = fileService
+    this.batchJobService_ = batchJobService
+    this.productService_ = productService
+    this.productVariantService_ = productVariantService
+    this.shippingProfileService_ = shippingProfileService
+    this.productRepo_ = productRepository
+    this.productOptionRepo_ = productOptionRepository
+    this.productVariantRepo_ = productVariantRepository
+    this.regionRepo_ = regionRepository
+  }
+
+  buildTemplate(): Promise<string> {
+    throw new Error("Not implemented!")
+  }
+
+  /**
+   * Create a description of a row on which the error occurred and throw a Medusa error.
+   *
+   * @param row - Parsed CSV row data
+   */
+  protected static throwDescriptiveError(row: TParsedRowData): never {
+    const message = `Error while processing row with:
+      product id: ${row["product.id"]},
+      product handle: ${row["product.handle"]},
+      variant id: ${row["variant.id"]}
+      variant sku: ${row["variant.sku"]}`
+
+    throw new MedusaError(MedusaError.Types.INVALID_DATA, message)
+  }
+
+  /**
+   * Generate instructions for update/create of products/variants from parsed CSV rows.
+   *
+   * @param csvData - An array of parsed CSV rows.
+   */
+  async getImportInstructions(
+    csvData: TParsedRowData[]
+  ): Promise<Record<OperationType, TParsedRowData[]>> {
+    const regionRepo = this.manager_.getCustomRepository(this.regionRepo_)
+
+    const shippingProfile = await this.shippingProfileService_.retrieveDefault()
+
+    const seenProducts = {}
+
+    const productsCreate: TParsedRowData[] = []
+    const productsUpdate: TParsedRowData[] = []
+
+    const variantsCreate: TParsedRowData[] = []
+    const variantsUpdate: TParsedRowData[] = []
+
+    for (const row of csvData) {
+      if ((row["variant.prices"] as object[]).length) {
+        await this.handleVariantPrices(row, regionRepo)
+      }
+
+      if (row["variant.id"]) {
+        variantsUpdate.push(row)
+      } else {
+        variantsCreate.push(row)
+      }
+
+      // save only first occurrence
+      if (!seenProducts[row["product.handle"] as string]) {
+        row["product.profile_id"] = shippingProfile
+        ;(row["product.product.id"] ? productsUpdate : productsCreate).push(row)
```

Comment:
suggestion: don't find that line pretty clear. Maybe in that case we can use an if statement instead :)